addon/ng2/utilities/find-parent-module.ts

```diff
@@ -0,0 +1,34 @@
+import * as fs from 'fs';
+import * as path from 'path';
+
+module.exports = function findParentModule(
+  project: any,
+  currentDir: string): string {
+
+  const sourceRoot = path.join(project.root, project.ngConfig.apps[0].root, 'app');
+
+  // trim currentDir
+  currentDir = currentDir.replace(path.join(project.ngConfig.apps[0].root, 'app'), '');
+
+  let parts = currentDir.split(path.sep);
+  const partsLength = parts.length;
+  let pathToCheck = path.join(sourceRoot, currentDir);
+
+
+  for (let i = 0; i <= partsLength; i++) {
+    let files: string[] = fs.readdirSync(pathToCheck);
```

Comment:
You don't need the type information, and you could just do

```ts
const files = fs.readdirSync(pathToCheck)
  .filter(fileName => fileName.endsWith('.module.ts')
  .filter(fileName => fs.statsSync(fileName).isFile());
```

Right now there's a bug (unlikely but still) that if it contains `.module.ts` (but doesn't end with) it will match. Also, a bug that it doesn't check if it's a directory or a regular file.